### ISSUE TITLE
Docker plugin doesn't log correctly CPU and Mem per container

### ISSUE THREAD

**Issue by mvaude** [influxdata/telegraf#2182]
### Relevant telegraf.conf:

```toml
# Read metrics about docker containers
[[inputs.docker]]
  ## Docker Endpoint
  ##   To use TCP, set endpoint = "tcp://[ip]:[port]"
  ##   To use environment variables (ie, docker-machine), set endpoint = "ENV"
  endpoint = "unix:///var/run/docker.sock"
  ## Only collect metrics for these containers, collect all if empty
  container_names = []
  ## Timeout for docker list, info, and stats commands
  timeout = "5s"

  ## Whether to report for each container per-device blkio (8:0, 8:1...) and
  ## network (eth0, eth1, ...) stats or not
  perdevice = true
  ## Whether to report for each container total blkio and network stats or not
  total = false
```

### System info:

- `Telegraf v1.1.1 (git: release-1.1.0 94de9dca1fc6efb3a4bf3ec6869c356278c6755a)`
- `Linux version 3.14.28-031428-generic (apw@tangerine) (gcc version 4.6.3 (Ubuntu/Linaro 4.6.3-1ubuntu5) ) #201501081937 SMP Thu Jan 8 19:39:13 UTC 2015`
- (circle CI container)

### Steps to reproduce:

1. run telegraf with docker input plugin in circle CI
2. run computation tasks in some containers
3. analyze output in influxdb (ex: through chronograf)

#### Query Example

```
SELECT "usage_percent" FROM "metrics"."autogen"."docker_container_cpu" WHERE time > now() - 30m GROUP BY "host", "container_name"
```

#### Expected result

- Values variation regarding containers activities.

### Actual behavior:

- Static values: 0

### Use case:

- Running telegraf locally or inside CI to be able to gather some metrics from each run.

**Comment by sparrc** [comment]
can you explain how you know the values are incorrect? what values are you comparing it to? what are the "correct" and incorrect values?

**Comment by mvaude** [comment]
I am running tests using containers and they are definitely using some CPU and MEM resources.
When I execute the query in InfluxDB, I can see some values but there are always equal to `0`.

The correct values should be some real CPU load instead of 0 during all the test suites run-time.

**Comment by sparrc** [comment]
haven't been able to reproduce this locally using an influxdb docker container and running influx-stress (on macOS)

```
> SELECT usage_percent,container_name FROM docker_container_cpu WHERE time > now() - 5m
name: docker_container_cpu
time			usage_percent		container_name
----			-------------		--------------
2016-12-20T13:43:30Z	0.042717587939698494	xenodochial_noether
2016-12-20T13:43:32Z	0.036621499999999994	xenodochial_noether
2016-12-20T13:43:34Z	0.04138703241895262	xenodochial_noether
2016-12-20T13:43:36Z	0.044813200000000004	xenodochial_noether
2016-12-20T13:43:38Z	0.0593289		xenodochial_noether
2016-12-20T13:43:40Z	0.5484208040201005	xenodochial_noether
2016-12-20T13:43:42Z	0.03364251256281407	xenodochial_noether
2016-12-20T13:43:44Z	0.034620551378446114	xenodochial_noether
2016-12-20T13:43:46Z	0.0364923		xenodochial_noether
2016-12-20T13:43:48Z	0.03255810473815461	xenodochial_noether
2016-12-20T13:43:50Z	0.029082		xenodochial_noether
2016-12-20T13:43:52Z	0.0454314		xenodochial_noether
2016-12-20T13:43:54Z	0.04162			xenodochial_noether
2016-12-20T13:43:56Z	0.0459692		xenodochial_noether
2016-12-20T13:43:58Z	0.04893216957605985	xenodochial_noether
2016-12-20T13:44:00Z	0.02601866004962779	xenodochial_noether
2016-12-20T13:44:02Z	0.0305249		xenodochial_noether
2016-12-20T13:44:04Z	0.0275027		xenodochial_noether
2016-12-20T13:44:06Z	0.31262215538847116	xenodochial_noether
2016-12-20T13:44:08Z	0.034140947630922694	xenodochial_noether
2016-12-20T13:44:10Z	0.05376382871536524	xenodochial_noether
2016-12-20T13:44:12Z	0.0373651		xenodochial_noether
2016-12-20T13:44:14Z	0.05132220551378446	xenodochial_noether
2016-12-20T13:44:16Z	0.0452251		xenodochial_noether
2016-12-20T13:44:18Z	0.04097142857142857	xenodochial_noether
2016-12-20T13:44:20Z	0.03615351620947631	xenodochial_noether
2016-12-20T13:44:22Z	0.04656481203007519	xenodochial_noether
2016-12-20T13:44:24Z	0.05184467661691543	xenodochial_noether
2016-12-20T13:44:26Z	0.0527446		xenodochial_noether
2016-12-20T13:44:28Z	0.033841608040201004	xenodochial_noether
2016-12-20T13:44:30Z	0.04663271820448878	xenodochial_noether
2016-12-20T13:44:32Z	0.036069874686716796	xenodochial_noether
2016-12-20T13:44:34Z	0.0306278		xenodochial_noether
2016-12-20T13:44:36Z	0.030770552763819098	xenodochial_noether
2016-12-20T13:44:38Z	0.14081869346733666	xenodochial_noether
2016-12-20T13:44:40Z	0.026342985074626867	xenodochial_noether
2016-12-20T13:44:42Z	0.03836			xenodochial_noether
2016-12-20T13:44:44Z	0.02704452261306533	xenodochial_noether
2016-12-20T13:44:46Z	0.0407939		xenodochial_noether
2016-12-20T13:44:48Z	0.023619497487437184	xenodochial_noether
2016-12-20T13:44:50Z	0.0461782		xenodochial_noether
2016-12-20T13:44:52Z	0.0437733		xenodochial_noether
2016-12-20T13:44:54Z	0.03193413533834586	xenodochial_noether
2016-12-20T13:44:56Z	0.029917799999999998	xenodochial_noether
2016-12-20T13:44:58Z	0.0510673		xenodochial_noether
2016-12-20T13:45:00Z	0.06565725685785537	xenodochial_noether
2016-12-20T13:45:02Z	0.05224493702770781	xenodochial_noether
2016-12-20T13:45:04Z	0.0408686		xenodochial_noether
2016-12-20T13:45:06Z	0.04622633416458853	xenodochial_noether
2016-12-20T13:45:08Z	0.0424338		xenodochial_noether
2016-12-20T13:45:10Z	0.03939501246882793	xenodochial_noether
2016-12-20T13:45:12Z	0.0657683		xenodochial_noether
2016-12-20T13:45:14Z	0.049728478802992526	xenodochial_noether
2016-12-20T13:45:16Z	0.08779208121827412	xenodochial_noether
2016-12-20T13:45:18Z	0.0562496		xenodochial_noether
2016-12-20T13:45:20Z	0.058044		xenodochial_noether
2016-12-20T13:45:22Z	0.044478195488721804	xenodochial_noether
2016-12-20T13:45:24Z	0.04511950124688279	xenodochial_noether
2016-12-20T13:45:26Z	0.03930034912718204	xenodochial_noether
2016-12-20T13:45:28Z	0.05976551378446115	xenodochial_noether
2016-12-20T13:45:30Z	0.042572200000000004	xenodochial_noether
2016-12-20T13:45:32Z	0.04768452261306533	xenodochial_noether
2016-12-20T13:45:34Z	0.20120791979949876	xenodochial_noether
2016-12-20T13:45:36Z	0.0504966		xenodochial_noether
2016-12-20T13:45:38Z	0.0427125		xenodochial_noether
2016-12-20T13:45:40Z	0.038393416458852865	xenodochial_noether
2016-12-20T13:45:42Z	0.04011990024937656	xenodochial_noether
2016-12-20T13:45:44Z	0.04058603491271821	xenodochial_noether
2016-12-20T13:45:46Z	0.061235112219451375	xenodochial_noether
2016-12-20T13:45:48Z	0.05332914572864321	xenodochial_noether
2016-12-20T13:45:50Z	0.038153		xenodochial_noether
2016-12-20T13:45:52Z	0.054906004962779156	xenodochial_noether
2016-12-20T13:45:54Z	0.12754496240601504	xenodochial_noether
2016-12-20T13:45:56Z	0.049141645885286786	xenodochial_noether
2016-12-20T13:45:58Z	0.0531385		xenodochial_noether
2016-12-20T13:46:00Z	0.03521785536159601	xenodochial_noether
2016-12-20T13:46:02Z	0.043131072319202	xenodochial_noether
2016-12-20T13:46:04Z	0.0398953		xenodochial_noether
2016-12-20T13:46:06Z	0.0380418		xenodochial_noether
2016-12-20T13:46:08Z	0.0438424		xenodochial_noether
2016-12-20T13:46:10Z	0.0435675		xenodochial_noether
2016-12-20T13:46:12Z	0.04714922693266833	xenodochial_noether
2016-12-20T13:46:14Z	0.0466581		xenodochial_noether
2016-12-20T13:46:16Z	0.042553		xenodochial_noether
2016-12-20T13:46:18Z	0.037202977667493795	xenodochial_noether
2016-12-20T13:46:20Z	0.05419930174563591	xenodochial_noether
2016-12-20T13:46:22Z	0.047550300000000004	xenodochial_noether
2016-12-20T13:46:24Z	0.04363047619047619	xenodochial_noether
2016-12-20T13:46:26Z	0.04092671679197995	xenodochial_noether
2016-12-20T13:46:28Z	0.3166563184079602	xenodochial_noether
2016-12-20T13:46:30Z	0.037126516290726816	xenodochial_noether
2016-12-20T13:46:32Z	0.04027749373433584	xenodochial_noether
2016-12-20T13:46:34Z	0.0371678		xenodochial_noether
2016-12-20T13:46:36Z	0.04788551378446115	xenodochial_noether
2016-12-20T13:46:38Z	0.0457696		xenodochial_noether
2016-12-20T13:46:40Z	0.03759639097744361	xenodochial_noether
2016-12-20T13:46:42Z	0.04539647355163728	xenodochial_noether
2016-12-20T13:46:44Z	0.0410743		xenodochial_noether
2016-12-20T13:46:46Z	0.0421327680798005	xenodochial_noether
2016-12-20T13:46:48Z	0.18599739348370928	xenodochial_noether
2016-12-20T13:46:50Z	0.0492202493765586	xenodochial_noether
2016-12-20T13:46:52Z	0.034102299999999995	xenodochial_noether
2016-12-20T13:46:54Z	0.03593042606516291	xenodochial_noether
2016-12-20T13:46:56Z	66.87609766233766	xenodochial_noether <--- Started influx-stress here
2016-12-20T13:46:58Z	30.858493903743316	xenodochial_noether
2016-12-20T13:47:00Z	22.0745684375		xenodochial_noether
2016-12-20T13:47:02Z	25.24492683937824	xenodochial_noether
2016-12-20T13:47:04Z	19.702711088082904	xenodochial_noether
2016-12-20T13:47:06Z	37.93332580310881	xenodochial_noether
2016-12-20T13:47:08Z	19.534615165394403	xenodochial_noether
2016-12-20T13:47:10Z	19.64455321336761	xenodochial_noether
2016-12-20T13:47:12Z	19.586312142857143	xenodochial_noether
2016-12-20T13:47:14Z	18.722489514066496	xenodochial_noether
```

can you reproduce on a regular linux machine? TBH I don't know if I would consider the CircleCI a "normal" environment worth supporting

**Comment by mvaude** [comment]
I will make some tests and let you know.
It is maybe because when running a build in circle CI, it is a container by itself.

**Comment by sdiaz** [comment]
What we see, is that values returned are lower than average, we have for example a container that have values with `docker stats <id_container>` from 0,3% to 86% but normally you should see more values upper than 30%, and instead telegraf is reporting values of 0,15-0,40

**Comment by danielnelson** [comment]
I believe this was fixed in #3043, can you retest with 1.4.4 or newer?
